[Swift in WebKit] The Swift implementation of TextExtraction types is incorrect

rr-codes opened the pull request:
#### 6c8ce53a001b3a2e951b7d3e0ac891fc4b691c43
<pre>
[Swift in WebKit] The Swift implementation of TextExtraction types is incorrect
<a href="https://bugs.webkit.org/show_bug.cgi?id=292422">https://bugs.webkit.org/show_bug.cgi?id=292422</a>
<a href="https://rdar.apple.com/150497419">rdar://150497419</a>

Reviewed by Aditya Keerthi and Wenson Hsieh.

Work towards being able to move Text Extraction code to WebKit proper.

The WKTextExtractionItem and related types were previously incorrectly implemented in Swift;
they were essentially using a form of forward-declarations, rather than using the language
support for Swift implementations of Obj-C interfaces (`@objc @implementation`), which offers
significant benefits such as:
1. Being built-in as a language feature and the intended way to implement this mechanism
2. Compile-time checking of types, so that the Obj-C interface and the Swift implementation are never out-of-sync
3. Removes redundant type declarations for enums
4. Removes the need to specify availability versions and make all Swift types `public`
5. Enforces consistent naming conventions

Fix by making all the relevant types adopt `@objc @implementation`.

* Source/WebKit/WebKitSwift/TextExtraction/WKTextExtractionItem.h:

- Use `NS_ASSUME_NONNULL_BEGIN` so that the null-ness of the types are consistent between Obj-C and Swift
- Add nullability annotations where needed
- Add a missing initializer declaration to `WKTextExtractionRequest`
- Remove the redundant definition of the `WKTextExtractionContainer` enum

* Source/WebKit/WebKitSwift/TextExtraction/WKTextExtractionItem.swift:

- Use `@objc @implementation` for all types and remove `@objc public` everywhere
- Remove all now-unnecessary availability annotations

(WKTextExtractionLink.url):

- Workaround a limitation of `@objc @implementation` (detailed in an inline comment)

(WKTextExtractionRequest.fulfill(_:)):

- Use the proper Swift spelling of this function

* Source/WebKit/WebKitSwift/TextExtraction/WKWebView+TextExtraction.swift:
(createEditable(for:)):

- Drive-by improvement: Use the standard `nil` value instead of `Optional.none`

(WKWebView._requestTextExtraction(in:completionHandler:)):

- Use the proper Swift spelling of this function

Canonical link: <a href="https://commits.webkit.org/294442@main">https://commits.webkit.org/294442@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/430189cc7a23163080f91a4e526ce292417f81eb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101837 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21505 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11821 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/106995 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/52471 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21813 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/30012 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/77539 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/52471 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104844 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/16854 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91954 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57877 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16681 "Passed tests") | | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/51822 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86534 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10049 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109352 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/28970 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/21337 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/86518 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29331 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88155 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86090 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21907 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30841 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8564 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/23141 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28898 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/34189 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/28709 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/32032 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30268 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/WebKitSwift/TextExtraction/WKTextExtractionItem.swift:
```diff
@@ -22,75 +22,81 @@
-    @objc public let url: NSURL
-    @objc public let range: NSRange
+@_objcImplementation extension WKTextExtractionLink {
+    // Used to workaround the fact that `@objc @implementation` does not support stored properties whose size can change
```
I think this comment needs to be updated, since we're not actually using `@objc @implementation` yet?

whsieh reviewed (approved):
r=me, as long as the tests are passing.

rr-codes committed:
[Swift in WebKit] The Swift implementation of TextExtraction types is incorrect
https://bugs.webkit.org/show_bug.cgi?id=292422
rdar://150497419

Reviewed by Aditya Keerthi and Wenson Hsieh.

Work towards being able to move Text Extraction code to WebKit proper.

The WKTextExtractionItem and related types were previously incorrectly implemented in Swift;
they were essentially using a form of forward-declarations, rather than using the language
support for Swift implementations of Obj-C interfaces (`@objc @implementation`), which offers
significant benefits such as:
1. Being built-in as a language feature and the intended way to implement this mechanism
2. Compile-time checking of types, so that the Obj-C interface and the Swift implementation are never out-of-sync
3. Removes redundant type declarations for enums
4. Removes the need to specify availability versions and make all Swift types `public`
5. Enforces consistent naming conventions

Fix by making all the relevant types adopt `@objc @implementation`.

* Source/WebKit/WebKitSwift/TextExtraction/WKTextExtractionItem.h:

- Use `NS_ASSUME_NONNULL_BEGIN` so that the null-ness of the types are consistent between Obj-C and Swift
- Add nullability annotations where needed
- Add a missing initializer declaration to `WKTextExtractionRequest`
- Remove the redundant definition of the `WKTextExtractionContainer` enum

* Source/WebKit/WebKitSwift/TextExtraction/WKTextExtractionItem.swift:

- Use `@objc @implementation` for all types and remove `@objc public` everywhere
- Remove all now-unnecessary availability annotations

(WKTextExtractionLink.url):

- Workaround a limitation of `@objc @implementation` (detailed in an inline comment)

(WKTextExtractionRequest.fulfill(_:)):

- Use the proper Swift spelling of this function

* Source/WebKit/WebKitSwift/TextExtraction/WKWebView+TextExtraction.swift:
(createEditable(for:)):

- Drive-by improvement: Use the standard `nil` value instead of `Optional.none`

(WKWebView._requestTextExtraction(in:completionHandler:)):

- Use the proper Swift spelling of this function

Canonical link: https://commits.webkit.org/294442@main
